lib/mix/tasks/upload.ex

```diff
@@ -72,43 +69,19 @@ defmodule Mix.Tasks.Upload do
     Uploading to #{ip}...
     """)
 
-    # Options:
-    #
-    # ConnectTimeout - don't wait forever to connect
-    # PreferredAuthentications=publickey - since keyboard interactivity doesn't
-    #                                     work, don't try password entry options.
-    # -T - No pseudoterminals since they're not needed for firmware updates
-    opts = [
-      :stream,
-      :binary,
-      :exit_status,
-      :hide,
-      :use_stdio,
-      {:args,
-       [
-         "-o",
-         "ConnectTimeout=3",
-         "-o",
-         "PreferredAuthentications=publickey",
-         "-T",
-         "-s",
-         ip,
-         "fwup"
-       ]},
-      {:env,
-       [
-         {~c"LD_LIBRARY_PATH", false}
-       ]}
-    ]
-
-    port = Port.open({:spawn_executable, ssh_path()}, opts)
-
-    fd = File.open!(firmware_path, [:read])
-
-    Process.flag(:trap_exit, true)
-
-    sender_pid = spawn_link(fn -> send_data(port, fd) end)
-    port_read(port, sender_pid)
+    # LD_LIBRARY_PATH is unset to avoid errors with host ssl (see commit 9b1df471)
+    {_, status} = InteractiveCmd.cmd("ssh", ["-s", ip, "fwup"], env: [{"LD_LIBRARY_PATH", false}])
```

Comment:
How does this actually send the file? It seems like it just opens for the Fwup subsystem, but we're no longer transferring bytes?

Reply:
lol, I figured out how in the world I tricked myself. Thanks for catching this. This is really funny. A fix will be coming in a bit.

Reply:
It works now or most accurately, it works on Circuits Quickstart firmware running in QEMU using a non-default SSH port number with a currently unsubmitted PR. Sorry about that.